src/Components/WebAssembly/WebAssembly/src/Hosting/MutableConfiguration.cs

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Hosting
+{
+    /// <summary>
+    /// MutableConfiguration is a class that implements the interface of an IConfiguration,
+    /// IConfigurationRoot, and IConfigurationBuilder. It can be used to simulatneously build
+    /// and read from a configuration object.
+    /// </summary>
+    public class MutableConfiguration : IConfiguration, IConfigurationRoot, IConfigurationBuilder
+    {
+        private readonly IList<IConfigurationProvider> _providers = new List<IConfigurationProvider>();
```

Comment:
minor nitpick - using a type like `IList<>` over `List<>` *for your implementation details like members* generally just has the effect of making your code slower and more complex. `List<>` has a richer API that `IList<>` and you're unlikely to change back in the other direction if you just build the habit of using `List<>`.

Reply:
Got it. I can change this.

Reply:
Fixed.